### PR TITLE
feat: added git database tasks for github integration

### LINKED ISSUE #201: [TRI-709] Add git database tasks to `@trigger.dev/github` integration
The `@trigger.dev/github` integration currently only supports a limited number of tasks found here:

https://github.com/triggerdotdev/trigger.dev/blob/main/integrations/github/src/tasks.ts

We should add tasks for cover the entire "git database" collection of rest API endpoints, found here:

- [ ] Blobs: https://docs.github.com/en/rest/git/blobs?apiVersion=2022-11-28
- [ ] Commits: https://docs.github.com/en/rest/git/commits?apiVersion=2022-11-28
- [ ] References: https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28
- [ ] Tags: https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28
- [ ] Trees: https://docs.github.com/en/rest/git/trees?apiVersion=2022-11-28

<sub>[TRI-709](https://linear.app/triggerdotdev/issue/TRI-709/add-git-database-tasks-to-triggerdevgithub-integration)</sub>

### PR DESCRIPTION
**Pull request by akhill10**
Fixes #201 

Added following git database tasks for github integration
- [x]  [Blobs](https://docs.github.com/en/rest/git/blobs?apiVersion=2022-11-28)
- [x]  [Commits](https://docs.github.com/en/rest/git/commits?apiVersion=2022-11-28)
- [x]  [References](https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28)
- [x]  [Tags](https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28)
- [x]  [Trees](https://docs.github.com/en/rest/git/trees?apiVersion=2022-11-28)

/claim #201

### CODE REVIEW

**Comment by ericallam** [comment]
This looks great! I'll do a proper review shortly 👍

File: integrations/github/src/tasks.ts [hunk 1]
```diff
@@ -604,6 +604,502 @@ const listOrgWebhooks: GithubAuthenticatedTask<
...
+        {
+          label: "Commit SHA",
+          text: params.commitSHA,
+        },
+      ],
+    };
+  },
+};
+
+type ListMatchinghReferencesTask = GithubAuthenticatedTask<
```
Comment (ericallam) [inline integrations/github/src/tasks.ts]: Typo here, should be `ListMatchingReferencesTask`

**Review by ericallam — CHANGES_REQUESTED** [review]
This looks close! It would be great to get an example usage of a few of these tasks in the [github.ts](./examples/nextjs-example/src/jobs/github.ts) file as well.

**Comment by akhill10** [comment]
> This looks close! It would be great to get an example usage of a few of these tasks in the [github.ts](https://github.com/triggerdotdev/trigger.dev/pull/examples/nextjs-example/src/jobs/github.ts) file as well.

Sure, will do that shortly.

**Comment by ericallam** [comment]
Examples look great, will be merging this in shortly 👍

**Review by ericallam — APPROVED** [review]

**Pull request merged**